### PR TITLE
Clean up some functions to be more in line with DRY

### PR DESCRIPTION
**Pull request by daankoning**
mostly small fixes

see commit messages

### CODE REVIEW

File: tabbycat/standings/ranking.py [hunk 1]
```diff
@@ -18,6 +18,18 @@
...
+def _generate_ordering(metrics, annotators, min_field, min_rounds):
+    ordering = []
+    annotations = {a.key: a for a in annotators}
+    for key in metrics:
+        annotation = annotations[key]
+        if annotation.ascending:
+            ordering.append(annotation.get_ranking_annotation(min_field, min_rounds).asc(nulls_last=True))
+        else:
+            ordering.append(annotation.get_ranking_annotation(min_field, min_rounds).desc(nulls_last=True))
+    return ordering
```
Comment (tienne-B) [inline tabbycat/standings/ranking.py:30]: How about have this as a `@staticmethod` inside `BaseRankAnnotator`, keeping the `_get_ordering()` name? Then we wouldn't need the methods to merely call this function. As a result, line 167 would look something like (would also need to modify the other places calling `_get_ordering()`):
```py
order_by=self._get_ordering(self.metrics[1:], annotators, min_field, min_rounds),
```

File: tabbycat/templates/allocations/DragAndDropStore.js [hunk 2]
```diff
@@ -249,20 +249,22 @@ export default new Vuex.Store({
...
-          panelClashesCombined[key].push(...value);
+          clashesOrHistoriesCombined[key].push(...value);
         }
       })
 
-      return panelClashesCombined
+      return clashesOrHistoriesCombined
+    },
+    panelClashesForItem: (state, getters) => (id) => {
+      return getters.panelClashesOrHistoriesForItem(id, 'clashes')
```
Comment (tienne-B) [inline tabbycat/templates/allocations/DragAndDropStore.js:267]: As you can tell, I'm less a fan of methods merely calling another method, and as used only in `HoverableConflictMixin.vue`, I'd be fine with something like this for `HoverableConflictMixin.vue:19-20`:
```py
clashes = this.panelClashesOrHistoriesForItem(itemId, 'clashes')
histories = this.panelClashesOrHistoriesForItem(itemId, 'histories')
```

File: tabbycat/adjallocation/conflicts.py [hunk 3]
```diff
@@ -235,7 +236,10 @@ def seen_adj_adj(self, adj1, adj2):
         covered by this object."""
         return (adj1.id, adj2.id) in self.adjadjhistories
 
-    def serialized_by_participant(self):
+    def serialized_by_participant(self) -> Tuple[
+        Dict[int, Dict[str, List[Dict[str, int]]]],
+        Dict[int, Dict[str, List[Dict[str, int]]]],
+    ]:
```
[the remark this answers is not in the document]
Comment (daankoning) [reply tabbycat/adjallocation/conflicts.py:242]: The purpose of these, to me isn't even entirely about human readability. For me, intellisense being able to parse these already makes life significantly easier and prevents some bugs.

 If you disagree just let me know :).

[the remark this answers is not in the document]
Comment (tienne-B) [reply tabbycat/adjallocation/conflicts.py:242]: But humans are the ones who'll end up having to read that mess! I think we could clean up the typing though by using two identical `TypedDict`s (one for adjs, other for teams).

[the remark this answers is not in the document]
Comment (daankoning) [reply tabbycat/adjallocation/conflicts.py]: I'm not entirely sure I implemented this as elegantly as you intended (as I'm not completely sure how you intend for this to look) but I had a stab in 11df5b5. Let me know what you think!

File: tabbycat/adjallocation/conflicts.py [hunk 4]
```diff
@@ -235,7 +249,7 @@ def seen_adj_adj(self, adj1, adj2):
         covered by this object."""
         return (adj1.id, adj2.id) in self.adjadjhistories
 
-    def serialized_by_participant(self):
+    def serialized_by_participant(self) -> Tuple[Dict[int, TeamConflicts], Dict[int, AdjudicatorConflicts]]:
```
[the remark this answers is not in the document]
Comment (tienne-B) [reply tabbycat/adjallocation/conflicts.py:251]: Oh, I see that it isn't well supported: https://stackoverflow.com/questions/65568841/how-to-make-a-typeddict-with-integer-keys

So it's fine without.

**Pull request merged**